File: ios-native/BuilderHQ/Features/Onboarding/Builder/WizardChrome.swift
/// WizardChrome — shared header / footer / progress views used by the
/// builder onboarding wizard. Kept in one file because each piece is
/// small and reads more clearly together than across three files.

import SwiftUI

// MARK: - Progress dots

/// Slim pill-and-dots progress indicator. The current step is a
/// horizontally elongated pill in accent; past steps are solid accent
/// dots; future steps are hairline dots. A floating "1 of 7" caption
/// to the right gives explicit count context without crowding the bar.
///
/// Animates the pill widening / contracting on step change, plus a
/// staggered fill on past steps so jumping multiple steps reads as
/// natural progression rather than a snap.
struct WizardProgress: View {
    let total: Int
    let current: Int   // 0-indexed

    var body: some View {
        HStack(spacing: 10) {
            HStack(spacing: 6) {
                ForEach(0..<total, id: \.self) { index in
                    Capsule()
                        .fill(fillColor(for: index))
                        .frame(
                            width: index == current ? 26 : 6,
                            height: 6
                        )
                        .shadow(
                            color: index == current ? Palette.accentGlow : .clear,
                            radius: 6
                        )
                        .animation(
                            .spring(response: 0.45, dampingFraction: 0.78),
                            value: current
                        )
                }
            }

            Text("\(current + 1) of \(total)")
                .font(.system(size: 11, weight: .semibold))
                .tracking(1.2)
                .foregroundStyle(Palette.textDim)
                .monospacedDigit()
        }
    }

    private func fillColor(for index: Int) -> Color {
        if index < current { return Palette.accent.opacity(0.85) }
        if index == current { return Palette.accent }
        return Palette.hairlineStrong
    }
}

// MARK: - Wizard step header

/// Standardised header block — step index, title, optional caption.
/// Uses the same single-Text concat the auth headlines use so the
/// italic accent kisses the line above.
struct WizardHeader: View {
    /// "01", "02", … shown small and accent-tinted.
    let index: String
    let kicker: String
    let title: String
    /// Italicised accent fragment displayed below `title`. Use a one-
    /// or two-word phrase (e.g. "your business.").
    let titleAccent: String
    let caption: String?

    init(
        index: String,
        kicker: String,
        title: String,
        titleAccent: String,
        caption: String? = nil
    ) {
        self.index = index
        self.kicker = kicker
        self.title = title
        self.titleAccent = titleAccent
        self.caption = caption
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 14) {
            // Small kicker row — index + step label
            HStack(spacing: 8) {
                Text(index)
                    .font(.system(
                        size: 11,
                        weight: .bold,
                        design: .monospaced
                    ))
                    .tracking(2)
                    .foregroundStyle(Palette.accent)
                Text("/")
                    .font(.system(size: 11, weight: .bold))
                    .foregroundStyle(Palette.textDim)
                Text(kicker.uppercased())
                    .font(.system(size: 11, weight: .bold))
                    .tracking(2)
                    .foregroundStyle(Palette.textDim)
            }

            (
                Text(title + "\n")
                    .font(Typography.ui(size: 38, weight: .medium))
                    .foregroundStyle(Palette.text)
                + Text(titleAccent)
                    .font(Typography.serifItalic(size: 38))
                    .foregroundStyle(Palette.accentLight)
            )
            .tracking(-0.5)
            .lineSpacing(-4)

            if let caption = caption {
                Text(caption)
                    .font(Typography.body)
                    .foregroundStyle(Palette.textMuted)
            }
        }
    }
}

// MARK: - Wizard footer (Back + Continue)

/// Sticky footer with two affordances:
///   · Back (chevron + "Back") — left side, hidden on step 1
///   · Continue / Submit pill   — right side, primary CTA
///
/// The pill morphs through idle → spinner → checkmark just like the
/// auth submit button so completing a step feels consistent with the
/// rest of the app.
struct WizardFooter: View {
    let canContinue: Bool
    let isSaving: Bool
    let continueLabel: String
    let isFinal: Bool
    let onBack: (() -> Void)?
    let onContinue: () -> Void

    init(
        canContinue: Bool,
        isSaving: Bool,
        continueLabel: String = "Continue",
        isFinal: Bool = false,
        onBack: (() -> Void)?,
        onContinue: @escaping () -> Void
    ) {
        self.canContinue = canContinue
        self.isSaving = isSaving
        self.continueLabel = continueLabel
        self.isFinal = isFinal
        self.onBack = onBack
        self.onContinue = onContinue
    }

    var body: some View {
        HStack(spacing: 12) {
            if let onBack = onBack {
                Press(haptic: .tap, action: onBack) {
                    HStack(spacing: 6) {
                        Image(systemName: "chevron.left")
                            .font(.system(size: 12, weight: .bold))
                        Text("Back")
                            .font(.system(size: 14, weight: .semibold))
                    }
                    .foregroundStyle(Palette.textMuted)
                    .padding(.horizontal, 18)
                    .frame(height: 52)
                    .background(
                        Capsule().fill(Palette.surface)
                    )
                    .overlay(
                        Capsule().stroke(Palette.hairline, lineWidth: 1)
                    )
                }
            }

            // Continue / Submit pill — fills remaining width.
            Button(action: onContinue) {
                ZStack {
                    Capsule()
                        .fill(canContinue ? Palette.accent : Palette.surfaceElev)
                        .shadow(
                            color: canContinue ? Palette.accentGlow : .clear,
                            radius: 22,
                            x: 0,
                            y: 10
                        )

                    if isSaving {
                        ProgressView()
                            .progressViewStyle(.circular)
                            .tint(Palette.accentContrast)
                            .transition(.opacity)
                    } else {
                        HStack(spacing: 10) {
                            Text(continueLabel)
                                .font(.system(size: 16, weight: .bold))
                            Image(systemName: isFinal ? "checkmark" : "arrow.right")
                                .font(.system(size: 14, weight: .bold))
                        }
                        .foregroundStyle(canContinue ? Palette.accentContrast : Palette.textDim)
                        .transition(.opacity)
                    }
                }
                .frame(height: 52)
                .frame(maxWidth: .infinity)
                .animation(Motion.easeOut, value: isSaving)
                .animation(Motion.easeOut, value: canContinue)
            }
            .disabled(!canContinue || isSaving)
            .sensoryFeedback(.impact(weight: .medium), trigger: isSaving)
        }
        .padding(.horizontal, 24)
        .padding(.bottom, 16)
        .padding(.top, 12)
        // No background — the footer is stacked beneath the step's
        // ScrollView (not overlaying it), so the AmbientBackground shows
        // through cleanly. A gradient here only painted a flat strip
        // across the bottom safe-area with nothing behind it to fade.
    }
}

// MARK: - Inline banner error (shared with auth flows)

struct WizardBanner: View {
    let message: String

    var body: some View {
        HStack(spacing: 10) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 13))
                .foregroundStyle(Palette.danger)
            Text(message)
                .font(.system(size: 13, weight: .medium))
                .foregroundStyle(Palette.text)
                .frame(maxWidth: .infinity, alignment: .leading)
        }
        .padding(.horizontal, 14)
        .padding(.vertical, 12)
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(Palette.dangerMuted)
        )
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(Palette.danger.opacity(0.35), lineWidth: 1)
        )
    }
}

// MARK: - Section block (used on multi-section steps)

/// Numbered subsection within a step (e.g. "02 / Business address",
/// "02 / Postal address" in step 2). Less heavy than WizardHeader.
struct WizardSectionLabel: View {
    let text: String

    var body: some View {
        Text(text.uppercased())
            .font(.system(size: 11, weight: .bold))
            .tracking(2.2)
            .foregroundStyle(Palette.textDim)
    }
}

File: android-native/app/src/main/java/au/com/builderhq/app/feature/onboarding/builder/WizardChrome.kt
package au.com.builderhq.app.feature.onboarding.builder

import androidx.compose.animation.animateColorAsState
import androidx.compose.animation.core.animateDpAsState
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.rounded.ArrowBack
import androidx.compose.material.icons.automirrored.rounded.ArrowForward
import androidx.compose.material.icons.rounded.Check
import androidx.compose.material.icons.rounded.WarningAmber
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import au.com.builderhq.app.core.design.components.pressable
import au.com.builderhq.app.core.design.theme.Bhq
import au.com.builderhq.app.core.design.theme.DisplaySerif

/**
 * Chrome for the builder onboarding wizard — progress capsules,
 * two-tone headline, footer nav, error banner, section labels.
 * Mirrors the iOS WizardChrome 1:1.
 */

/** Row of 7 capsules; current = elongated accent pill. "1 of 7" right. */
@Composable
fun WizardProgress(total: Int, current: Int, modifier: Modifier = Modifier) {
    val c = Bhq.colors
    Row(modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
        Row(Modifier.weight(1f), horizontalArrangement = Arrangement.spacedBy(5.dp)) {
            repeat(total) { i ->
                val width by animateDpAsState(
                    if (i == current) 26.dp else 8.dp,
                    spring(dampingRatio = 0.85f),
                    label = "capsuleW",
                )
                val color by animateColorAsState(
                    when {
                        i == current -> c.accent
                        i < current -> c.accent.copy(alpha = 0.85f)
                        else -> c.blueprintLine.copy(alpha = 0.14f)
                    },
                    label = "capsuleC",
                )
                Box(
                    Modifier
                        .width(width)
                        .height(6.dp)
                        .clip(CircleShape)
                        .background(color),
                )
            }
        }
        Text(
            "${current + 1} of $total",
            color = c.textDim,
            fontSize = 12.sp,
            fontWeight = FontWeight.Medium,
        )
    }
}

/**
 * The standard step headline: `{index} / {KICKER}` eyebrow, then a
 * two-tone title — [title] in the UI face and [titleAccent] in the
 * serif italic brand face — plus an optional muted caption.
 */
@Composable
fun WizardHeader(
    index: String,
    kicker: String,
    title: String,
    titleAccent: String,
    caption: String? = null,
) {
    val c = Bhq.colors
    Column {
        Text(
            "$index / ${kicker.uppercase()}",
            color = c.accentLight,
            fontSize = 11.sp,
            fontWeight = FontWeight.Bold,
            letterSpacing = 2.sp,
        )
        Spacer(Modifier.height(10.dp))
        Text(
            buildAnnotatedString {
                append(title)
                append("\n")
                withStyle(
                    SpanStyle(
                        fontFamily = DisplaySerif,
                        fontStyle = FontStyle.Italic,
                        fontWeight = FontWeight.Normal,
                        color = c.accentLight,
                    ),
                ) { append(titleAccent) }
            },
            color = c.text,
            fontSize = 34.sp,
            fontWeight = FontWeight.SemiBold,
            lineHeight = 40.sp,
        )
        if (caption != null) {
            Spacer(Modifier.height(10.dp))
            Text(caption, color = c.textMuted, fontSize = 14.sp, lineHeight = 20.sp)
        }
    }
}

/** UPPERCASE dim section label for the numbered subsections. */
@Composable
fun WizardSectionLabel(text: String, modifier: Modifier = Modifier) {
    Text(
        text.uppercase(),
        color = Bhq.colors.textDim,
        fontSize = 11.sp,
        fontWeight = FontWeight.Bold,
        letterSpacing = 2.2.sp,
        modifier = modifier,
    )
}

/** Danger banner for VM bannerError. */
@Composable
fun WizardBanner(message: String, modifier: Modifier = Modifier) {
    val c = Bhq.colors
    Row(
        modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(c.danger.copy(alpha = 0.10f))
            .border(1.dp, c.danger.copy(alpha = 0.28f), RoundedCornerShape(12.dp))
            .padding(horizontal = 14.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(Icons.Rounded.WarningAmber, null, tint = c.danger, modifier = Modifier.size(16.dp))
        Spacer(Modifier.width(10.dp))
        Text(message, color = c.danger, fontSize = 13.sp, lineHeight = 18.sp)
    }
}

/**
 * Footer nav: ghost Back pill (hidden when [onBack] is null — step 1)
 * + primary continue pill that morphs idle → spinner → label. The
 * final step's pill shows a check instead of an arrow.
 */
@Composable
fun WizardFooter(
    canContinue: Boolean,
    isSaving: Boolean,
    onContinue: () -> Unit,
    modifier: Modifier = Modifier,
    continueLabel: String = "Continue",
    isFinal: Boolean = false,
    onBack: (() -> Unit)? = null,
) {
    val c = Bhq.colors
    Row(modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
        if (onBack != null) {
            Row(
                Modifier
                    .clip(CircleShape)
                    .background(c.fillSubtle)
                    .pressable(onClick = onBack)
                    .padding(horizontal = 18.dp, vertical = 13.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Icon(
                    Icons.AutoMirrored.Rounded.ArrowBack, null,
                    tint = c.textMuted, modifier = Modifier.size(16.dp),
                )
                Spacer(Modifier.width(6.dp))
                Text("Back", color = c.textMuted, fontSize = 15.sp, fontWeight = FontWeight.Medium)
            }
            Spacer(Modifier.width(12.dp))
        }
        val enabled = canContinue && !isSaving
        Row(
            Modifier
                .weight(1f)
                .clip(CircleShape)
                .background(if (enabled) c.accent else c.accent.copy(alpha = 0.35f))
                .pressable(enabled = enabled, onClick = onContinue)
                .padding(vertical = 15.dp),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (isSaving) {
                CircularProgressIndicator(
                    color = c.accentContrast, strokeWidth = 2.dp,
                    modifier = Modifier.size(18.dp),
                )
            } else {
                Text(
                    continueLabel,
                    color = c.accentContrast,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                )
                Spacer(Modifier.width(8.dp))
                Icon(
                    if (isFinal) Icons.Rounded.Check else Icons.AutoMirrored.Rounded.ArrowForward,
                    null,
                    tint = c.accentContrast,
                    modifier = Modifier.size(18.dp),
                )
            }
        }
    }
}
